kafka_consumer_freshness_tracker/src/test/java/com/tesla/data/consumer/freshness/ConsumerFreshnessTest.java

```diff
@@ -272,6 +277,156 @@ public void testBurrowMissingConsumerGroupPartitionEndOffsetMarksMissing() throw
     });
   }
 
+  @Test
+  public void testConfigurationContainsUnknownCluster() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "bad_cluster";
+    when(burrow.getClusterDetail(clusterName)).thenThrow(new IOException());
+    Map<String, Object> conf = mockConfForCluster(clusterName, "l1.example.com:9092, l2.example.com:9092, l3.example.com:9092");
+
+    withExecutor(executor -> {
+      ConsumerFreshness freshness = new ConsumerFreshness();
+      freshness.setupWithBurrow(conf, burrow);
+
+      assertEquals("Should be no worker queue for unknown cluster", 0, freshness.getAvailableWorkersForTesting().size(),
+          0.0);
+
+      FreshnessMetrics metrics = freshness.getMetricsForTesting();
+      assertEquals("Should be a log for failed Burrow.getClusterDetail request", 1,
+          metrics.burrowClusterDetailReadFailed.get(), 0.0);
+    });
+  }
+
+
+  @Test
+  public void testNormalModeConfigurationIsValidServersMatchBurrow() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterDetail(clusterName))
+      .thenReturn(mockBurrowClusterDetailResponse(
+            Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251")
+            )
+          );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251, kafka02.example.com:10251, kafka03.example.com:10251"
+        );
+
+    withExecutor(executor -> {
+      ConsumerFreshness freshness = new ConsumerFreshness();
+
+      // This exception should be thrown when the consumers are attempted to be
+      // connected after validation succeeds
+      thrown.expect(KafkaException.class);
+      thrown.expectMessage("Failed to construct kafka consumer");
+
+      freshness.setupWithBurrow(conf, burrow);
+
+      assertEquals("Should be a worker queue for valid cluster", 1, freshness.getAvailableWorkersForTesting().size(),
```

Comment:
is this line even reachable if its thrown in the setup?

Reply:
removed